include/veriblock/blockchain/chain.hpp

```diff
@@ -57,9 +57,11 @@ struct Chain {
     return (*this)[index->height + 1];
   }
 
-  height_t size() const { return (height_t)chain.size() + startHeight_; }
+  height_t height() const { return (height_t)chain.size() + startHeight_ - 1; }
```

Comment:
when chain is empty, and startHeight is 0, height will point to -1, which is wrong. Better use `size() -1` in places where you need `height` of the chain

Reply:
I am not sure, because the meaning of the size is not the similar with the realisation, it is not return the size of the chain.